src/test/lib/when_test.ts

```diff
@@ -0,0 +1,123 @@
+/**
+ * @license
+ * Copyright (c) 2018 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+/// <reference path="../../../node_modules/@types/mocha/index.d.ts" />
+/// <reference path="../../../node_modules/@types/chai/index.d.ts" />
+
+import {html, render} from '../../core.js';
+import {when} from '../../lib/when.js';
+import {stripExpressionDelimeters} from '../test-helpers.js';
+
+const assert = chai.assert;
+
+suite('when', () => {
+  let container: HTMLDivElement;
+
+  setup(() => {
+    container = document.createElement('div');
+  });
+
+  suite('simple template', () => {
+    function renderWhen(condition: boolean) {
+      render(
+        html`${when(condition, () => html`<div></div>`, () => html`<span></span>`)}`,
+        container
+      );
+    }
+
+    test('renders if/then template based on condition', () => {
+      renderWhen(true);
+      assert.equal(stripExpressionDelimeters(container.innerHTML), '<div></div>');
+
+      renderWhen(false);
+      assert.equal(stripExpressionDelimeters(container.innerHTML), '<span></span>');
+
+      renderWhen(true);
+      assert.equal(stripExpressionDelimeters(container.innerHTML), '<div></div>');
+    });
+
+    test('caches if/then templates', () => {
+      let ifEl;
```

Comment:
these can be const

Reply:
These are assigned values later on, so they can't be a const.